terraform.tf

```diff
@@ -54,7 +54,7 @@ terraform {
   required_providers {
     azurerm = {
       source  = "hashicorp/azurerm"
-      version = "3.113.0"
+      version = "~> 4.0.0"
```

Comment:
configuration in RD components ensured that not only patch releases were pulled in automatically, but minor releases as well. this has worked reliably for about a year now. is there a reason to restrict automatic updates to patch only? If not, recommend to change to "~> 4.0"